platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/DefaultSignedStateSentinel.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.signed;
+
+import static com.swirlds.logging.legacy.LogMarker.EXCEPTION;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.utility.CompareTo;
+import com.swirlds.common.utility.RuntimeObjectRecord;
+import com.swirlds.common.utility.RuntimeObjectRegistry;
+import com.swirlds.common.utility.throttle.RateLimiter;
+import com.swirlds.platform.config.StateConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * This object is responsible for observing the lifespans of signed states, and taking action if a state suspected of a
+ * memory leak is observed.
+ */
+public class DefaultSignedStateSentinel implements SignedStateSentinel {
+
+    private static final Logger logger = LogManager.getLogger(DefaultSignedStateSentinel.class);
+
+    private final Time time;
+    private final RateLimiter rateLimiter;
+
+    private final Duration maxSignedStateAge;
+
+    /**
+     * Create an object that monitors signed state lifespans.
+     *
+     * @param platformContext the current platform's context
+     */
+    public DefaultSignedStateSentinel(@NonNull final PlatformContext platformContext) {
+        this.time = platformContext.getTime();
+        maxSignedStateAge = platformContext
+                .getConfiguration()
+                .getConfigData(StateConfig.class)
+                .suspiciousSignedStateAge();
+
+        rateLimiter = new RateLimiter(Time.getCurrent(), Duration.ofMinutes(10));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void checkSignedStates() {
+        final RuntimeObjectRecord objectRecord = RuntimeObjectRegistry.getOldestActiveObjectRecord(SignedState.class);
```

Comment:
would it make sense to add a check `rateLimiter.request()` *before* fetching object records?

No sense in doing extra computation if we're in the cool down period and won't write any logs anyway

Reply:
added check that no-ops fast.